packages/nodes-base/nodes/HttpRequest/HttpRequest.node.ts

```diff
@@ -1121,3 +1371,56 @@ export class HttpRequest implements INodeType {
 		return this.prepareOutputData(returnItems);
 	}
 }
+
+const NODES_BASE_ROOT: Readonly<string> = path.resolve(__dirname, '..', '..', '..');
+
+/**
+ * Credential types shown as options for `Node Credential Type` parameter.
+ */
+const CREDENTIAL_TYPES = getCredPaths().reduce<INodePropertyOptions[]>((acc, credPath) => {
+	const credential = new (getCredClass(credPath))();
+
+	if (!isSupportedByHttpRequestNode(credential)) return acc;
+
+	return [
+		...acc,
+		{
+			name: credential.displayName,
+			value: credential.name,
+		},
+	];
+}, []);
+
+function getCredPaths(root = NODES_BASE_ROOT): string[] {
+	const packageJson = require(path.resolve(root, 'package.json'));
+
+	return deduplicate(packageJson.n8n.credentials);
```

Comment:
Prevent Vue unique key complains from [duplicates in `package.json`](https://github.com/n8n-io/n8n/pull/3215/files#diff-595c8c8a9ed3c0a98b04046183607efac6bb434fc5e755c8cae6aee50c280f76).